compiler/rustc_typeck/src/check/check.rs

```diff
@@ -494,18 +494,17 @@ pub(super) fn check_opaque_for_inheriting_lifetimes(
                 InternalSubsts::identity_for_item(tcx, def_id.to_def_id()),
             ),
             generics: tcx.generics_of(def_id),
-            ty: None,
         };
         let prohibit_opaque = tcx
             .explicit_item_bounds(def_id)
             .iter()
-            .any(|(predicate, _)| predicate.visit_with(&mut visitor).is_break());
+            .try_for_each(|(predicate, _)| predicate.visit_with(&mut visitor));
         debug!(
             "check_opaque_for_inheriting_lifetimes: prohibit_opaque={:?}, visitor={:?}",
             prohibit_opaque, visitor
         );
 
-        if prohibit_opaque {
+        if let Some(ty) = prohibit_opaque.break_value() {
```

Comment:
Can `ty.unwrap()` fail here? I would expect this to not be the case. Might be missing something though.

Maybe if we have a `PredicateAtom::RegionOutlives` :thinking:

Reply:
IIUC it can: https://github.com/rust-lang/rust/blob/149f593bf6d204ea226461c18a95f558a6ad250b/compiler/rustc_typeck/src/check/check.rs#L462-L473

Reply:
yeah, but this only ends up returning `None` if we aren't inside of a type, as we would otherwise return `Some(ty)` there